### PR TITLE
[2] fix: Add error handling for CircuitBreaker timeout

### PR DESCRIPTION
**Pull request by ibu1224**
## Context

When a timeout occurs in CircuitBreaker, err.status becomes undefined.
Now the following code will be processed successfully.
https://github.com/screwdriver-cd/screwdriver/blob/f1b25fd35576310e9119393cb890ffbb625f1317/plugins/webhooks/index.js#L416-L421

This will cause problems in later processing, Add the case when a timeout occurs in CircuitBreaker.

It also adds information on which pipeline the error occurred.

## Objective

In this PR, I will add error handling when CircuitBreaker times out in [getCommitSha()](https://github.com/screwdriver-cd/scm-github/blob/4c2a843ce5266a14b27d4788b3cd6d9b8f6e98a3/index.js#L860).

Circuitbreaker will output a log like the following when it times out.
```
Failed to getCommitSha:  CircuitBreaker timeout
```

~~Error handling is done with the `CircuitBreaker timeout` string in err.message. Is there a better way?~~
Fix circuit breaker to return error status 504 on timeout

## References
[1] https://github.com/screwdriver-cd/circuit-fuses/pull/30

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/webhooks/index.js [hunk 1]
```diff
@@ -887,6 +891,8 @@ async function createEvents(
                 } catch (err) {
                     if (err.status >= 500) {
                         throw err;
+                    } else if (err.message !== undefined && err.message.indexOf('CircuitBreaker timeout') !== -1) {
```
Comment (wahapo) [inline plugins/webhooks/index.js:894]: How about using "if `err.status` is `undefined`"?

Comment (jithine) [inline plugins/webhooks/index.js:894]: what's the status code for curcuit breaker errors ? perhaps we should define something at https://github.com/screwdriver-cd/circuit-fuses/blob/d99a1b3679803624f941eb8668ca2274eda37393/index.js to make that clear.

Also the `else` flow of simply logging seems to handle a valid scenario which is not well defined.

Comment (ibu1224) [reply plugins/webhooks/index.js:894]: You're right.
I'll change the way I fix it.

Comment (ibu1224) [reply plugins/webhooks/index.js]: @jithine 
Fixed by the following PR.
https://github.com/screwdriver-cd/circuit-fuses/pull/30

File: plugins/webhooks/index.js [hunk 2]
```diff
@@ -542,7 +542,7 @@ async function pullRequestOpened(options, request, h) {
             return h.response().code(201);
         })
         .catch(err => {
-            logger.error(`[${hookId}]: ${err}`);
+            logger.error(`[${hookId}, pipeline:${options.pipeline.id}]: ${err}`);
```
Comment (yoshwata) [inline plugins/webhooks/index.js:545]: Don't you touch pullRequestSync and pullRequestClosed?

Comment (yoshwata) [reply plugins/webhooks/index.js:545]: I prefer such logs that shows where it occurs.
スクリーンショット 2021-04-26 15 18 21

**Review by wahapo — APPROVED** [review]

**Review by yoshwata — APPROVED** [review]

**Review by tkyi — APPROVED** [review]

**Pull request merged**